oasislmf/manager.py

```diff
@@ -417,6 +418,67 @@ def generate_oasis_files(
             exposure_fp, _keys_fp, exposure_profile=exposure_profile
         )
 
+        # Fill summary of exposure data dictionary
+        def fill_exposure_summary(
+            exposure_summary_df,
+            exposure_summary,
+            peril,
+            status,
+        ):
+
+            for coverage_type in SUPPORTED_COVERAGE_TYPES:
```

Comment:
@hchagani-oasislmf I will review the PR fully later, but this Pandas code will have to be refactored to perform operations on columns and sections, rather than doing loops over rows. Various other changes will be required, especially in how columns and subsets of columns are accessed.